code/headers/moving_platform.hpp

```diff
@@ -0,0 +1,65 @@
+#pragma once
+#include "hwlib.hpp"
+#include <base_module.hpp>
+
+namespace r2d2::moving_platform {
+	
+	/**
+	 * Class moving_platform_c is an abstract class that can be implemented for all different kinds of motors
+	 * This class exists so you can use a moving platform even without knowing what kind of platform you have been provided with. 
+	 */
+    class moving_platform_c : public base_module_c {
+    protected:
+        /**
+		 * this speed represents a throtle in percentages (%)
+		 * This means that 100 is max and forward, -100 is backwards
+		 * 
+		 * the angle is represented in degrees
+		 */
+        int8_t speed, angle;
+
+		/**
+		 * set the speed to the given value
+		 */
+        virtual void set_speed(const int8_t &speed) = 0;
```

Comment:
Is a int8_t large enough for the motor speed? Are we losing to much precision on this?